File: doc-examples/nitrite-example-java/src/main/java/example/TimestampedRecord.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

// tag::lifecycle-entity[]
@MappedEntity
public class TimestampedRecord {
    @Id
    @GeneratedValue
    private String id;

    private String name;

    public TimestampedRecord() {
    }

    public TimestampedRecord(String name) {
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
// end::lifecycle-entity[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/TimestampedRecord.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

// tag::lifecycle-entity[]
@MappedEntity
class TimestampedRecord {
    @Id
    @GeneratedValue
    var id: String? = null

    var name: String? = null

    constructor()

    constructor(name: String) {
        this.name = name
    }
}
// end::lifecycle-entity[]
